File: tests/test_s0001_two_sum.py
import pytest
from ..src.solutions.s0001_two_sum import Solution

def test_two_sum():
    solution = Solution()
    assert solution.twoSum([2, 7, 11, 15], 9) == [0, 1]
    assert solution.twoSum([3, 2, 4], 6) == [1, 2]
    assert solution.twoSum([3, 3], 6) == [0, 1]

File: tests/test_s0001_two_sum.cpp
#include <gtest/gtest.h>

#include "../src/solutions/s0001_two_sum.cpp"

TEST(TwoSumTest, TestCases) {
  Solution solution;
  std::vector<int> result;

  result = solution.twoSum({2, 7, 11, 15}, 9);
  EXPECT_EQ(result, std::vector<int>({0, 1}));

  result = solution.twoSum({3, 2, 4}, 6);
  EXPECT_EQ(result, std::vector<int>({1, 2}));

  result = solution.twoSum({3, 3}, 6);
  EXPECT_EQ(result, std::vector<int>({0, 1}));
}
